src/subclass/container.rs

```diff
@@ -109,6 +114,16 @@ impl<T: ContainerImpl + ObjectImpl> ContainerImplExt for T {
             from_glib_none(f(container.to_glib_none().0, widget.to_glib_none().0))
         }
     }
+
+    fn parent_forall<P: FnMut(&Widget)>(&self, container: &Container, callback: P) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkContainerClass;
+            if let Some(f) = (*parent_class).forall {
+                f(container.to_glib_none().0, false, callback.to_glib_none().0)
```

Comment:
I'm currently struggling with this line. 

`pub forall: Option<unsafe extern "C" fn(*mut GtkContainer, gboolean, GtkCallback, gpointer)>,`

1) What does the gboolean here?!
2) How I'm supposed to handle the `gpointer`?

Reply:
> 1. What does the gboolean here?!

It switches between the `foreach` and `forall` behaviour. It's called `include_internals` and you would simply add that to all your functions here and pass it through. It's something the implementer has to take care of.

> 2. How I'm supposed to handle the `gpointer`?

That's the actually tricky part here. You get the `gpointer` from C (in `container_forall()` below). You would have to wrap that together with the `GtkCallback` in some new type:

```rust
pub struct Callback {
    cb: gtk_sys::GtkCallback,
    user_data: gpointer,
}

impl Callback {
    pub fn call(&self, widget: &Widget) {
        unsafe {
            self.cb(widget.to_glib_none().0, self.user_data);
        }
    }
}
```

And then pass such a `&Callback` through the Rust functions.

Reply:
It complains about `self.cb(...)`

```
error[E0599]: no method named `cb` found for reference `&subclass::container::Callback` in the current scope
   --> src/subclass/container.rs:269:18
    |
269 |             self.cb(widget.to_glib_none().0, self.user_data);
    |                  ^^----------------------------------------- help: remove the arguments
    |                  |
    |                  field, not a method
```

Reply:
`(self.cb)(...)` would be the solution here